src/bin/nydus-image/core/blob.rs

```diff
@@ -45,7 +45,10 @@ impl Blob {
                         blob_ctx.blob_readahead_size += size;
                     }
                 }
-                self.dump_meta_data(blob_ctx)?;
+                // Dump is only required if there is chunk in the blob
+                if blob_ctx.compressed_blob_size > 0 {
+                    self.dump_meta_data(blob_ctx)?;
```

Comment:
Move the `if blob_ctx.compressed_blob_size > 0` check into `dump_meta_data` should be better.

Reply:
fixed